app/services/incorrect_profile_remover.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# A class to remove profiles on policies that do not have a matching ref_id
+class IncorrectProfileRemover
+  def self.run!
+    Policy.includes(:profiles).find_each do |policy|
```

Comment:
Observation: The `.includes(:profiles)` is redundant, as those prefetched values/models are not used.

Could the prefetched values be used to do heuristics to improve run on a bigger data set? For example adding a condition `policy.profiles.count > 1` to prevent `policy.initial_profile.ref_id` doing a query for a single-profile policy.

Reply:
Removed the `includes`. It seems like a good idea to limit the run to policies with more than one profile, but the query is not so straightforward; because of that, I think it kind of falls under the category of premature optimization. There are currently less than 1500 policies today.

Reply:
My preference is to keep the migration within database and short as possible. That is using minimal queries as necessary (i.e. subqueries).

Reply:
The thing you suggested is not a SQL query, though: `policy.profiles.count > 1`. I took it as illustration only, but I don't know how to limit the policy list in a single query by that condition. Do you have a recommendation?

Reply:
Sorry, I see what you mean now. I'll keep the `includes` and add the `policy.profiles.count > 1` check. That _shouldn't_ add any other queries. :+1: thanks for the recommendation!

Reply:
I've meant it as a general rule for writing/executing migrations. My first idea was to use Ruby to limit number of queries made. There are ways to have it all in db, but that would likely break from the ORM and it is not necessary to do. :wink: